tests/sessions_tests/tests.py

```diff
@@ -597,33 +597,32 @@ def test_non_default_cache(self):
 
 class SessionMiddlewareTests(TestCase):
 
+    def run_middleware(self, request=None, response=None, modifier=None):
```

Comment:
This is an example of a method that could have an alternate implementation in a subclass that calls `process_request` and `process_response` directly, in order to test the middleware both old-style and new-style. Or maybe that's only worth it for middleware where we're making deeper changes in the conversion?